core/src/checks/verify_no_rollback.c

```diff
@@ -62,3 +71,26 @@ int verify_no_rollback(void) {
   INFO("verify_no_rollback: ok");
   return 0;
 }
+
+/**
+ * This hard-coded test case verifies that no_rollback_write_to_buf() produces a known output from the current
+ * VERSION_MAGIC and VERSION constants. It needs to be updated when the VERSION is incremented.
+ */
+int verify_no_rollback_write_to_buf(void) {
+  char buf[VERSION_SIZE];
+  char expected_buf[VERSION_SIZE] = { 0 };
+  static_assert(sizeof(buf) == sizeof(expected_buf), "buf and expected_buf should have the same size");
+
+  no_rollback_write_to_buf(VERSION_MAGIC, VERSION, buf);
+
+  // Note: the VERSION string is at the end. Update it when updating the VERSION constant.
+  const char* expected_string = "8414-210";
```

Comment:
feels a bit annoying to have to manually update this with each test addition, but I don't yet understand the rationale behind why we need this

Reply:
Can we use the VERSION #define directly or does that defeat the purpose of the test?

Reply:
It defeats the purpose of the test I think. This is a hard-coded test which verifies that for known inputs the result is a known output. There's only 1 more place that needs to be changed, it will cause self-checks to fail immediately, and there are very visible comments so I think it's gonna be ok.